pkg/plugins/runtime/k8s/controllers/gateway_instance_controller.go

```diff
@@ -273,33 +281,72 @@ func (r *GatewayInstanceReconciler) createOrUpdateDeployment(
 				secContext.Capabilities.Add = append(secContext.Capabilities.Add, "NET_BIND_SERVICE")
 			}
 
+			if fsGroup := gatewayInstance.Spec.PodTemplate.Spec.PodSecurityContext.FSGroup; fsGroup != 0 {
+				if securityContext == nil {
+					securityContext = &kube_core.PodSecurityContext{}
+				}
+				securityContext.FSGroup = &fsGroup
+			}
+
+			if readOnlyRFS := gatewayInstance.Spec.PodTemplate.Spec.Container.SecurityContext.ReadOnlyRootFilesystem; readOnlyRFS {
+				container.SecurityContext.ReadOnlyRootFilesystem = &readOnlyRFS
+			}
+
+			container.VolumeMounts = []kube_core.VolumeMount{
+				{
+					Name:      "tmp-dir",
```

Comment:
nit but this is its name in the egress/ingress objects

```suggestion
					Name:      "tmp",
```